### PR TITLE
a11y: add ARIA labels to various buttons and fields

### LINKED ISSUE #2117: [Accessibility][Name, Role, Value - Property Pane_Ask a question (Bot Asks)]: Name is not defined for up and down arrow in 'Max Turn count' spin button.
**User Experience** 
Users who rely on AT will might get confused if the controls name is not defined for the available controls.

**Test Environment**
Browser: Microsoft Edge Canary Version: 82.0.425.0 (Official build)
OS build: 2004 (19564.1005)
Screen Reader: Narrator
URL: [Bot Framework Composer](http://localhost:3000/dialogs/Main)
[**Prerequisite Here**](https://github.com/microsoft/BotFramework-Composer)

**Repro Steps:** 
1. Open the above mentioned URL in EDGE and turn on narrator.
2. Navigate to already created dialog tree item and activate it. Such as- Greetings, Actions, EndDialog etc.
3. Click on "+" and select 'Ask a question'. Then select any of the available options except Oauth login.
4. Navigate to property pane appeared at right side of the screen.
5. Now, navigate to 'Max turn count' spin button.
6. Observe the issue.

**Actual:** 
The name is not defined for the up/down arrow in the 'Max Turn count' spin button.
Narrator only narrates their role as button on getting the focus in scan mode.

**Expected:**
The name should be defined for the controls.
Narrator should announce increase and decrease button.

**Note:**
Same issue observed at every property pane which have spin button such as "Connect to QnA Knowledgebase".

**MAS Impacted:** [MAS4.1.2](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730})

**Attachment:**
MAS4 1 2_Name is not defined for up and down key

### LINKED ISSUE #2125: [Accessibility][Name,Role,value - About]: Ensures buttons have discernible text.
[edit by @corinagum] 
The issues filed below have been addressed, but there are a couple other related problems that I would like addressed in this issue. To quote my comment [below](https://github.com/microsoft/BotFramework-Composer/issues/2125#issuecomment-613142023): 

> [R]elated to this PR, I believe the about page needs to be modified to remove `tabIndex={-1}`, and the `<IconBotton>` should be changed to just an icon inside `<Link>`

Once the above is addressed, we can return this to 'waiting for verification'

---

**User Experience:**
This issue should be resolved for good coding practice if buttons have discernible text.

**Test Environment:**
Environment: Chromium Edge (Anaheim) + Narrator
Build Version: 2004 (OS Build 19564.1005)​​
Microsoft Edge Version: 82.0.425.0 (Official build) canary (64-bit)​
Test URL: http://localhost:3000/home

**Repro Steps:**
1. Open URL: http://localhost:3000/home
2. "Bot Framework Composer" homepage will gets opened.
3. Select any of the previously created bot from "Recent Bots" or create one from the available Bots under 'Examples' and activate it.
4. Select 'About' from left navigation.
5. Run K4W tool on the page and Observe the issue

**Actual Result:**
'Terms of Use' and 'Privacy' buttons does not have discernible text. Screen reader read only as button.

This accessibility issue was found using Accessibility Insights for Web(keros), a tool that helps debug and find accessibility issues earlier. Get more information & download this tool at http://aka.ms/Keros.

**Expected Result:**
Ensures buttons i.e. 'Terms of Use' and 'Privacy' have discernible text. Screen reader should read the name of the button.

**Note:**
Same issue observed for 'Design Flow' pane.
Same issue observe throughout the website for every property pane. {Attached some of screenshot below, please look the issue throughout the application}

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730}

**Attachment:**
[MAS4.1.2_About_K4W.zip](https://github.com/microsoft/BotFramework-Composer/files/4266576/MAS4.1.2_About_K4W.zip)
ensures button have discernible text

### LINKED ISSUE #2140: [Accessibility][Label and Instruction - Property Pane_Ask a question (Bot Asks)]: Ensures every form element has a label
**User Experience** 
Users who rely on AT will face difficulty to locate the focus if the alt attribute is not defined for the image.

**Test Environment**
Browser: Microsoft Edge Canary Version: 82.0.425.0 (Official build)
OS build: 2004 (19564.1005)
URL: [Bot Framework Composer](http://localhost:3000/dialogs/Main)
[**Prerequisite Here**](https://github.com/microsoft/BotFramework-Composer)

**Repro Steps:** 
1. Open the above mentioned URL in EDGE.
2. "Bot Framework Composer" homepage will gets opened.
3. Select any of the previously created bot from "Recent Bots" or create one from the available Bots under 'Examples' and activate it.
4. Navigate to already created dialog tree item and activate it. Such as- Greetings, Actions, EndDialog etc.
5. Click on "+" and select 'Ask a question'. Then select any of the available options except Oauth login.
6. Navigate to property pane appeared at right side of the screen and select 'Bot Asks'.
7. Run Accessibility insight and observe the issue.

**Actual:** 
Bot Framework logo image doesn't have alternate text or a role of none or presentation.

**Expected:**
Ensures Bot Framework logo image have alternate text or a role of none or presentation.

**Note:**
This accessibility issue was found using Accessibility Insights for Web 2.14.1 (axe-core 3.3.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at [http://aka.ms/AccessibilityInsights](http://aka.ms/AccessibilityInsights).

**Observation:**
Same issue observe throughout the website for every property pane.

**Element path:**
.css-vaviek;#TextField942

**Snippet:**
`<input type="text" id="TextField942" placeholder="ActionsSample-0111" autocomplete="off" class="ms-TextField-field field-157" aria-invalid="false" value="ActionsSample-0111">`

**How to fix:**
    Fix any of the following:
      aria-label attribute does not exist or is empty
      aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
      Form element does not have an implicit (wrapped) <label>
      Form element does not have an explicit <label>
      Element has no title attribute or the title attribute is empty

**MAS Impacted:** [MAS3.3.2](https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={4b95a70d-258c-474a-820e-5c8b79d5c0bf})

**Attachment:**
Element must have label
Element must have labelother

### PR DESCRIPTION
**Pull request by beyackle**
## Description

This adds aria-label properties to many of the components that were missing them; this should take care of the linked a11y issues.

## Task Item

Closes #2117 
Closes #2125 
Closes #2140

### CODE REVIEW

File: Composer/packages/extensions/adaptive-form/src/components/fields/EditableField.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface EditableFieldProps extends FieldProps {
   fontSize?: string;
   styles?: Partial<ITextFieldStyles>;
   transparentBorder?: boolean;
+  ariaLabel?: string;
```
Comment (corinagum) [inline Composer/packages/extensions/adaptive-form/src/components/fields/EditableField.tsx:14]: I'm leaning towards making this a required label. what do you think?

File: data.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (a-b-r-o-w-n) [inline data.json:1]: whoops. this should not be committed.

File: Composer/packages/extensions/adaptive-form/src/components/FieldLabel.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ const DescriptionCallout: React.FC<DescriptionCalloutProps> = function Descripti
       }}
     >
       <IconButton
-        aria-labelledby={`${id}-description`}
+        aria-label={title + ':' + description}
```
Comment (corinagum) [inline Composer/packages/extensions/adaptive-form/src/components/FieldLabel.tsx:47]: does this need a space? `': '`

File: Composer/packages/extensions/adaptive-form/src/components/fields/EditableField.tsx (hunk 1 above)
Comment (beyackle) [reply Composer/packages/extensions/adaptive-form/src/components/fields/EditableField.tsx:14]: I agree it should be, but trying to do that causes all kinds of type-checking issues. I think that should be its own PR later on.

File: Composer/packages/extensions/adaptive-form/src/components/FieldLabel.tsx (hunk 3 above)
Comment (beyackle) [reply Composer/packages/extensions/adaptive-form/src/components/FieldLabel.tsx:47]: It works fine without one.

**Comment by corinagum** [comment]
@a-b-r-o-w-n Could you dismiss your review? `data.json` has been removed. :)

**Pull request merged**